version: "3.7"

volumes:
  acme: { driver: local }

services:
  reverse-proxy:
    image: traefik:${TRAEFIK_RELEASE}
    command:
      # DEBUG, INFO, WARN, ERROR, FATAL, PANIC
      --log.level=DEBUG
      --providers.docker=true
      --providers.docker.exposedbydefault=false
      --entryPoints.web.address=:80
      --entryPoints.websecure.address=:443
      --entrypoints._ENTRYPOINT_.http.redirections.entryPoint.to=_REDIRECT_
      --entrypoints._ENTRYPOINT_.http.redirections.entryPoint.scheme=_SCHEME_
      --certificatesresolvers.letsencrypt.acme.email=${ACME_EMAIL}
      --certificatesresolvers.letsencrypt.acme.storage=/letsencrypt/acme.json
      --certificatesresolvers.letsencrypt.acme.httpchallenge.entrypoint=websecure
      --certificatesresolvers.letsencrypt.acme.tlschallenge=_TLS_CHALLENGE_
      #--certificatesresolvers.letsencrypt.acme.caserver=https://acme-staging-v02.api.letsencrypt.org/directory
    ports:
      - "${HTTP_PORT}:80"
      - "${HTTPS_PORT}:443"
    volumes:
      - /var/run/docker.sock:/var/run/docker.sock
      - acme:/letsencrypt/
    restart: ${DOCKER_RESTART_POLICY}
    _NETWORK_ALIASES_

networks:
  antentafm:
